src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -144,7 +146,7 @@ public FormValidation doCheckUrl(@AncestorInPath Item project,
                 return FormValidation.ok();
 
             // get git executable on master
-            final EnvVars environment = new EnvVars(System.getenv()); // GitUtils.getPollEnvironment(project, null, launcher, TaskListener.NULL, false);
+            final EnvVars environment = project.getEnvironment(Jenkins.getActiveInstance(), TaskListener.NULL);
```

Comment:
I don't get it. What make you say we're getting git tool configuration for a remote system ? Jenkins.instance == master, where git client created on line 151 is ran.